modules/renter/contractor/contracts.go

```diff
@@ -194,6 +182,12 @@ func (c *Contractor) managedNewContract(host modules.HostDBEntry, contractFundin
 		return modules.RenterContract{}, err
 	}
 
+	// Add a mapping from the contract's id to the public key of the host.
+	c.mu.Lock()
+	c.contractIDToPubKey[contract.ID] = contract.HostPublicKey
+	c.pubKeysToContractID[string(contract.HostPublicKey.Key)] = contract.ID
+	c.mu.Unlock()
```

Comment:
should probably sanity check that you aren't obliterating a previous mapping, otherwise you are going to lose a contract and potentially not realize it.